utils/liftover.py

```diff
@@ -11,74 +11,122 @@
+            'gs://hail-common/references/human_g1k_v37.fasta.fai'
+            ) 
+    else:
+        source = hl.get_reference('GRCh37')
+        target = hl.get_reference('GRCh38')
+        chain = 'gs://hail-common/references/grch37_to_grch38.over.chain.gz'
+        target.add_sequence(
+            'gs://hail-common/references/Homo_sapiens_assembly38.fasta.gz', 
+            'gs://hail-common/references/Homo_sapiens_assembly38.fasta.fai'
+            )
+
+    source.add_liftover(chain, target)
+    return [source, target]
+
+
+def lift_data(t: Union[hl.MatrixTable, hl.Table], gnomad: bool, data_type: str, path: str, rg: hl.genetics.ReferenceGenome) -> Union[hl.MatrixTable, hl.Table]:
+    """
+    Lifts input Table or MatrixTable from one reference build to another
+
+    :param Table/MatrixTable t: Table or MatrixTable
     :param bool gnomad: Whether data is gnomAD data
     :param str data_type: Data type (exomes or genomes for gnomAD; not used otherwise)
-    :param str table_path: Path to input Table (if data is not gnomAD data)
+    :param str path: Path to input Table/MatrixTable (if data is not gnomAD data)
     :param ReferenceGenome rg: Reference genome
-    :return: Table with liftover annotations
-    :rtype: Table
+    :return: Table or MatrixTablewith liftover annotations
+    :rtype: Table or MatrixTable
     """
 
-    # annotate release file with liftover coordinates
-    ht = ht.annotate(new_locus=hl.liftover(ht.locus, rg, include_strand=True),
-                     old_locus=ht.locus
-                    )
-    no_target = ht.aggregate(hl.agg.count_where(hl.is_missing(ht.new_locus)))
-    logger.info('Filtering out {} sites that failed to liftover'.format(no_target))
-    ht = ht.filter(hl.is_defined(ht.new_locus)) 
-    ht = ht.key_by(locus=ht.new_locus.result, alleles=ht.alleles)
-    ht = ht.checkpoint(get_checkpoint_path(gnomad, data_type, table_path), overwrite=True)
-    return ht
+    logger.info('Annotating input with liftover coordinates')
+    liftover_expr = {
+        'new_locus': hl.liftover(t.locus, rg, include_strand=True),
+        'old_locus': t.locus
+    }
+    t = t.annotate(**liftover_expr) if isinstance(t, hl.Table) else t.annotate_rows(**liftover_expr)
+
+    no_target_expr = hl.agg.count_where(hl.is_missing(t.new_locus))
+    num_no_target = t.aggregate(no_target_expr) if isinstance(t, hl.Table) else t.aggregate_rows(no_target_expr)
+    
+    logger.info(f'Filtering out {num_no_target} sites that failed to liftover')
+    keep_expr = hl.is_defined(t.new_locus)
+    t = t.filter(keep_expr) if isinstance(t, hl.Table) else t.filter_rows(keep_expr)
+    
+    row_key_expr = {
+        'locus': t.new_locus.result,
+        'alleles': t.alleles
+    }
+    t = t.key_by(**row_key_expr) if isinstance(t, hl.Table) else t.key_rows_by(**row_key_expr)
+
+    logger.info('Writing out lifted over data')
+    t = t.checkpoint(get_checkpoint_path(gnomad, data_type, path, isinstance(t, hl.Table)), overwrite=True)
```

Comment:
as a general stylistic thing, you may want to parameterize `overwrite` and have that something you pass into the function (so that someone running it down the line doesn't accidentally overwrite a release file)